pandas/core/indexes/datetimelike.py

```diff
@@ -575,41 +564,30 @@ def shift(self, periods=1, freq=None):
     # --------------------------------------------------------------------
     # List-like Methods
 
-    def delete(self, loc):
-        new_i8s = np.delete(self.asi8, loc)
-
+    def _get_delete_freq(self, loc: int):
+        """
+        Find the `freq` for self.delete(loc).
+        """
         freq = None
         if is_period_dtype(self.dtype):
             freq = self.freq
-        elif is_integer(loc):
-            if loc in (0, -len(self), -1, len(self) - 1):
-                freq = self.freq
-        else:
-            if is_list_like(loc):
-                loc = lib.maybe_indices_to_slice(
-                    np.asarray(loc, dtype=np.intp), len(self)
-                )
-            if isinstance(loc, slice) and loc.step in (1, None):
-                if loc.start in (0, None) or loc.stop in (len(self), None):
+        elif self.freq is not None:
```

Comment:
i am not sure that these specific freq routines are really worth it given the extra code. Can't we just re-infer the freq? (sure if we do this constantly its not as performant, but this is not likey a common case).

ok for here, but if you can have a look at some point

Reply:
thats a reasonable thought.  i recently found a PITA bug in `infer_freq` though, so definitely not until that is sorted out

Reply:
kk maybe worthile to create an issue to see what implications / address later is.